core/user/PasswordConstraintsService.php

```diff
@@ -124,14 +124,19 @@ protected function register($config)
 
     /**
      * Any errors that was found during validation process
+     *
+     * @param string $password
+     *
      * @return array
      */
-    public function getErrors()
+    public function getErrors($password = '')
     {
         $errors = array();
         /** @var \tao_helpers_form_Validator $validator */
         foreach ($this->validators as $validator) {
-            $errors[] = $validator->getMessage();
+            if ( ! $validator->evaluate($password)) {
```

Comment:
it may be better to fill the error array in the validate method. So we have a method to validate, that returns true or false and another that returns errors without having to re-validate